File: 01_multiples_of_3_or_5/main.c
#include "stdio.h"

int sum(int start, int end);

int main(int argc, int** argv) {
  int test = sum(1, 10);
  int result = sum(1, 1000);
  
  printf("test: %d\n", test);
  printf("result: %d\n", result);

  return 0;
}

int sum(int start, int end) {
  int sum = 0;
  for (int i = start; i < end; i++) {
    if (i % 3 == 0 || i % 5 == 0) {
      sum += i;
    }
  }
  return sum;
}

File: 01_multiples_of_3_or_5/main.rs
fn main() {
    let test = sum(1, 10);
    println!("test: {:?}", test);

    let result = sum(1, 1000);
    println!("result: {:?}", result);
}

fn sum(start: i32, end: i32) -> i32 {
    let mut sum: i32 = 0;

    for n in start..end {
        if n % 3 == 0 || n % 5 == 0 {
            sum += n;
        }
    }

    sum
}
